### PR TITLE
[benchmark] Janitor Duty: Sweep III

### PR DESCRIPTION
**Pull request by palimondo**
Clean up setup overhead in few `Dictionary` benchmarks.

### CODE REVIEW

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
Agh… this is wrong. I need to fix this first. Sorry...

File: benchmark/single-source/DictionaryRemove.swift [hunk 1]
```diff
@@ -16,39 +16,26 @@ import TestsUtils
...
 
+let size = 100
+let numberMap = Dictionary(uniqueKeysWithValues: zip(1...size, 1...size))
+let boxedNums = (1...size).lazy.map { Box($0) }
+let boxedNumMap = Dictionary(uniqueKeysWithValues: zip(boxedNums, boxedNums))
+
 public let DictionaryRemove = [
   BenchmarkInfo(name: "DictionaryRemove",
-    runFunction: run_DictionaryRemove, tags: t, legacyFactor: 10),
+    runFunction: { for _ in 1...$0*100 {
```
Comment (eeckstein) [inline benchmark/single-source/DictionaryRemove.swift]: Please keep the benchmark code into a separate run-function. This makes it much easier to extract the benchmark into an isolated test case.

**Review by eeckstein — CHANGES_REQUESTED** [review]

File: benchmark/single-source/DictionaryRemove.swift (hunk 1 above)
Comment (palimondo) [reply benchmark/single-source/DictionaryRemove.swift]: I think it makes sense to have separate run function when it is reused between various benchmarks. The current order of attributes in `BenchmarkInfo` is a bit unfortunate for what I want to do here. I'll write up a short proposal to forum to get broader feedback, but I'd like to clean it up into taking trailing closure for the run function like this:

````swift
Benchmark("Name",
  aka: "OldName", legacyFactor: 10,
  setUp: { … }, tearDown: { … }) {
  for _ in 1...100 {
    workload
  }
}

Benchmark("Name",
  info: "Optional one line description of the benchmarks purpose ") {
  for _ in 1...100 {
    workload
  }
}

Benchmark("Name", setUp: { prepareWorkload(size: 123}, run: doTheStuff)
// assuming func doTheStuf is the run function shared between multiple benchmarks
````

There are few moving parts that I'd like to get feedback on, but my idea is to:
*  remove `tags` in favor of optional `skip: Bool`, as I think tags are over-engineered and unused in practice
* we should  add an `info` parameter after the `name` for a one-line description as was suggested in the Benchmark Naming Convention discussion.

---- 

Do you still want me to extract the closures into functions here, when they are not reused?

Comment (palimondo) [reply benchmark/single-source/DictionaryRemove.swift]: I have extracted the run functions.

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please benchmark and smoke test

**Comment by palimondo** [comment]
@swift-ci please benchmark and smoke test

**Comment by palimondo** [comment]
@swift-ci please benchmark and smoke test

**Comment by palimondo** [comment]
@swift-ci please bechmark

**Comment by palimondo** [comment]
@swift-ci please bechmark

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
@swift-ci please smoke test os x platform

**Comment by palimondo** [comment]
@swift-ci please smoke test os x platform

**Comment by palimondo** [comment]
@swift-ci please clean smoke test os x platform

**Comment by palimondo** [comment]
@swift-ci please clean smoke test os x platform

**Comment by palimondo** [comment]
@eeckstein: 😕 Should I re-adjust the `DictionarySwapAtOfObjects` by 10: `legacyFactor: 4`, loop multiplier `2500`? It seem it's no down to 64 μs in `-O`, that should clear the reported improvement.

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please benchmark

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
@swift-ci please smoke test

**Comment by palimondo** [comment]
@swift-ci please smoke test Linux platform

**Pull request merged**